src/routes/auth/+server.ts

```diff
@@ -11,62 +14,56 @@ const parseError = (error: string, component: string) => {
   return `Loading ${component}: ${error}`;
 }
 
-export async function POST({ request }: { request: Request }) {
-  const { apiKey, apiSecret } = await request.json();
-
-  if (!apiKey || !apiSecret) {
-    return json({ success: false, error: 'Missing API key or API secret' });
+const awaitRequestOrFailWith = async <T>(requestPromise: Promise<T>, component?: string) => {
+  try {
+    return await requestPromise
+  } catch(error) {
+    if (component) throw new Error(parseError((error as Error).message, component))
```

Comment:
It seems `component` is always provided by the caller.  If so, we could make `component` not optional, then we could remove the `else` condition.